charts/nexus3/files/repo.groovy

```diff
@@ -2,41 +2,36 @@ import groovy.json.JsonSlurper
 import groovy.json.JsonOutput
 import org.sonatype.nexus.repository.config.Configuration
 
-def params = new JsonSlurper().parseText(args)
-
-def repo = repository.repositoryManager.get(params.name)
-Configuration conf
+def repositoryManager = repository.repositoryManager
 
-// Cleanup input
+def params = new JsonSlurper().parseText(args)
 if (params.attributes?.cleanup?.policyName) {
   params.attributes.cleanup.policyName = params.attributes.cleanup.policyName.toSet()
 }
 
-// Check if repo already exists
-if (repo == null) {
-  // Create new repo
-  conf = new Configuration(
-    repositoryName: params.name,
-    recipeName: params.type,
-    online: params.online,
-    attributes: params.attributes
-  )
-  repository.createRepository(conf)
+def existingRepository = repositoryManager.get(params.name)
+Configuration configuration
+if (existingRepository == null) {
+  configuration = repositoryManager.newConfiguration()
+  configuration.repositoryName = params.name
+  configuration.recipeName = params.type
+  configuration.online = params.online
+  configuration.attributes = params.attributes
 } else {
-  // Get repo config
-  conf = repo.getConfiguration()
-  // println conf.inspect()
-  // println params.attributes.inspect()
+  configuration = existingRepository.getConfiguration()
 
-  // Test repo type isn't chaniging
-  if (conf.getRecipeName() != params.type) {
+  if (configuration.getRecipeName() != params.type) {
     throw new Exception("Tried to change recipe for repo ${params.name} to ${params.type}")
   }
 
-  // Update repo
-  conf.setOnline(params.online)
-  conf.setAttributes(params.attributes)
-  repo.stop()
-  repo.update(conf)
-  repo.start()
+  configuration.setOnline(params.online)
+  configuration.setAttributes(params.attributes)
```

Comment:
This needs to be wrapped in a `if (params.attributes) ...`. Otherwise we cannot set the online status of an existing repository without changing its attributes.

Reply:
It should work if you set the correct attributes for the repo you want to take offline.

Reply:
@stevehipwell that is true, but this means I would have to "redefine" all the default repositories I just want to disable. So instead of
```
  repos:
    # disable default repositories
    - name: "maven-central"
      online: false
    - name: "maven-public"
      online: false
    - name: "maven-releases"
      online: false
    - name: "maven-snapshots"
      online: false
    - name: "nuget-group"
      online: false
    - name: "nuget-hosted"
      online: false
    - name: "nuget.org-proxy"
      online: false
```
this would become pretty messy (or at least require a `attributes: []` which I consider a bit hacky and I am not sure would work in the first place)

Reply:
@stevehipwell I opened a pull request to address this

Reply:
Thanks, I've merged this in.